sa/storage-authority.go

```diff
@@ -673,6 +673,125 @@ func (ssa *SQLStorageAuthority) FinalizeAuthorization(authz core.Authorization)
 	return
 }
 
+const collectAuthIDs = `SELECT id FROM %s
+WHERE identifier = :ident AND
+status != :invalid AND
+status != :revoked AND
+expires > :now`
+
+// RevokeAuthorizationsByDomain invalidates all pending or finalized authorization
+// and any associated challenges for a single domain
+func (ssa *SQLStorageAuthority) RevokeAuthorizationsByDomain(ident core.AcmeIdentifier) (int64, int64, error) {
```

Comment:
Could you elaborate on what you were thinking for looping in authorization revocations? I'm not really sure how we'd do it beyond just looping over each authorization ID and trying to revoke it individually.